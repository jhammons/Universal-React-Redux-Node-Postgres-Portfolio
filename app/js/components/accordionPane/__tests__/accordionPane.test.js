import AccordionPane from '../accordionPane';

describe('<AccordionPane />', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<AccordionPane />);
    });

    test('renders without crashing', () => {
        expect(wrapper.find('.accordion')).toHaveLength(1);
    });

    test('has the right initial state', () => {
        expect(wrapper.state('toggleStatus')).toEqual('closed');
    });

    test('it renders children', () => {
        wrapper = shallow(
            <AccordionPane>
                <div>stuff</div>
            </AccordionPane>
        );
        expect(wrapper.contains(<div>stuff</div>)).toEqual(true);
        expect(wrapper).toMatchSnapshot();
    });

    test('it changes the toggleState when clicked', () => {
        wrapper = shallow(
            <AccordionPane>
                <div>stuff</div>
            </AccordionPane>
        );
        wrapper.find('.accordion').simulate('click');
        expect(wrapper.state('toggleStatus')).toEqual('open');
        expect(wrapper).toMatchSnapshot();

        wrapper.find('.accordion').simulate('click');
        expect(wrapper.state('toggleStatus')).toEqual('closed');
        expect(wrapper).toMatchSnapshot();
    });
});