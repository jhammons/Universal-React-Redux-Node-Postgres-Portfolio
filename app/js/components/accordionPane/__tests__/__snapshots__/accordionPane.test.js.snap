// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordionPane /> it changes the toggleState when clicked 1`] = `
<div
  className="accordion open"
  onClick={[Function]}
>
  <div>
    stuff
  </div>
</div>
`;

exports[`<AccordionPane /> it changes the toggleState when clicked 2`] = `
<div
  className="accordion closed"
  onClick={[Function]}
>
  <div>
    stuff
  </div>
</div>
`;

exports[`<AccordionPane /> it renders children 1`] = `
<div
  className="accordion closed"
  onClick={[Function]}
>
  <div>
    stuff
  </div>
</div>
`;
